src/main/java/technology/dice/dicefairlink/iterators/RandomisedCyclicIterator.java

```diff
@@ -34,6 +35,17 @@ public boolean hasNext() {
     return !elements.isEmpty();
   }
 
+  public boolean hasSameContent(final Collection<T> externalElements) {
+    if (externalElements == null || externalElements.size() != elements.size()) {
+      return false;
+    }
+    return externalElements.stream().unordered().parallel().allMatch(foundInInternalCollection());
```

Comment:
As discussed this is unnecessary given we're talking about collections of 2 or three elements. Just use: `new HashSet(externalElements).equals(new HashSet(elements))`.

Reply:
Original operation is more generic and does not require memory allocation/deallocation. Plus it can handle correctly all kind of collections.

Reply:
Original operation is more generic and does not require memory allocation/deallocation. Plus it can handle correctly all kind of collections.